lunatik.h

```diff
@@ -239,10 +240,13 @@ static inline T checker(lua_State *L, int ix)			\
 	return (T)object->private;				\
 }
 
-#define LUNATIK_PRIVATECHECKER(checker, T)			\
+#define LUNATIK_PRIVATECHECKER(checker, T, monitor)		\
 static inline T checker(lua_State *L, int ix)			\
 {								\
-	T private = (T)lunatik_toobject(L, ix)->private;	\
+	lunatik_object_t *object = likely(monitor) ?		\
+		lunatik_toobject(L, ix) :			\
+		lunatik_checkobject(L, ix);			\
+	T private = (T)object->private;				\
```

Comment:
I'm needing some similar right now and I realized it's not thread safe when you are not under a monitor.. because the object can be closed (that's, `->private = NULL` between `checkobject()` and `private =`.. I think we would need monitor on luacompletion anyways..

Reply:
see https://github.com/luainkernel/lunatik/blob/master/lunatik_obj.c#L89

Reply:
however, if not using `close`, I think there's no other flow that can set `private` to NULL.. it must be safe for this particular use case, I think.. I'm just not sure that we should provide this API.. it's kinda error prone IMO.. what do you think?

Reply:
to give you more context.. that's what I'm doing right now..

```
        lunatik_object_t *runtime = lunatik_checkobject(L, 1);

        lunatik_lock(runtime); /* now it's safe to check private */
        lua_State *Lr = (lua_State *)runtime->private;
        if (Lr == NULL)
                goto err;
        ...
   ```